cmd/protoc-gen-cpp-tableau-loader/helper/helper.go

```diff
@@ -43,10 +44,32 @@ func protocVersion(gen *protogen.Plugin) string {
 func NeedGenOrderedMap(md protoreflect.MessageDescriptor) bool {
 	opts := md.Options().(*descriptorpb.MessageOptions)
 	wsOpts := proto.GetExtension(opts, tableaupb.E_Worksheet).(*tableaupb.WorksheetOptions)
-	if wsOpts == nil || !wsOpts.OrderedMap {
+	if !wsOpts.GetOrderedMap() {
 		// Not an ordered map.
 		return false
 	}
+	if languages, ok := wsOpts.GetLangOptions()["OrderedMap"]; ok {
+		if !slices.Contains(strings.Split(languages, " "), "cpp") {
+			// Do not generate cpp ordered map
+			return false
+		}
+	}
+	return true
+}
+
+func NeedGenIndex(md protoreflect.MessageDescriptor) bool {
+	opts := md.Options().(*descriptorpb.MessageOptions)
+	wsOpts := proto.GetExtension(opts, tableaupb.E_Worksheet).(*tableaupb.WorksheetOptions)
+	if len(wsOpts.GetIndex()) == 0 {
+		// No index.
+		return false
+	}
+	if languages, ok := wsOpts.GetLangOptions()["Index"]; ok {
+		if !slices.Contains(strings.Split(languages, " "), "cpp") {
+			// Do not generate cpp index
+			return false
+		}
+	}
```

Comment:
Please add lang constants to `internal/options/options.go`:
```
const (
    optionOrderedMap = "OrderedMap"
    optionIndex = "Index"
)

const (
    LangCPP = "cpp"
    LangGO = "go"
)
```

Encapsulate func `NeedGenOrderedMap` and `NeedGenIndex`  in `internal/options/options.go` to check whether to generate or not.